unlock-app/src/components/content/CheckoutContent.tsx

```diff
@@ -1,165 +1,93 @@
 import CheckoutWrapper from '../interface/checkout/CheckoutWrapper'
 import CheckoutContainer from '../interface/checkout/CheckoutContainer'
-import { MetadataForm } from '../interface/checkout/MetadataForm'
 import { CheckoutErrors } from '../interface/checkout/CheckoutErrors'
 import { NotLoggedIn } from '../interface/checkout/NotLoggedIn'
+import { Locks } from '../interface/checkout/Locks'
+import Loading from '../interface/Loading'
+import { resetError } from '../../actions/error'
 import {
   Account as AccountType,
   Router,
   PaywallConfig,
-  UserMetadata,
 } from '../../unlockTypes'
 import getConfigFromSearch from '../../utils/getConfigFromSearch'
-import { useCheckoutCommunication } from '../../hooks/useCheckoutCommunication'
-import {
-  useCheckoutStore,
-  CheckoutStoreProvider,
-} from '../../hooks/useCheckoutStore'
-
-import {
-  setConfig,
-  setShowingLogin,
-  setShowingMetadataForm,
-} from '../../utils/checkoutActions'
-import { useSetUserMetadata } from '../../hooks/useSetUserMetadata'
-import { UnlockError } from '../../utils/Error'
-import { resetError } from '../../actions/error'
+import { CheckoutStoreProvider } from '../../hooks/useCheckoutStore'
 
 interface CheckoutContentProps {
   account: AccountType
   configFromSearch?: PaywallConfig
   errors: UnlockError[]
 }
 
-const defaultLockAddresses: string[] = []
-
-// This component wraps CheckoutContentInner so that it has access to the store.
 export const CheckoutContent = ({
-  account,
-  configFromSearch,
   errors,
-}: CheckoutContentProps) => {
-  return (
-    <CheckoutStoreProvider>
-      <CheckoutContentInner
-        account={account}
-        configFromSearch={configFromSearch}
-        errors={errors}
-      />
-    </CheckoutStoreProvider>
-  )
-}
-
-export const CheckoutContentInner = ({
-  account,
   configFromSearch,
-  errors,
+  account,
 }: CheckoutContentProps) => {
-  const reduxDispatch = useDispatch()
-  const { state, dispatch } = useCheckoutStore()
-  const { setUserMetadata } = useSetUserMetadata()
-  const { showingLogin, config, showingMetadataForm, delayedPurchase } = state
-
   const {
-    emitTransactionInfo,
     emitCloseModal,
-    emitUserInfo,
-  } = useCheckoutCommunication({
-    setConfig: (config: PaywallConfig) => {
-      dispatch(setConfig(config))
-    },
-  })
+    emitTransactionInfo,
+    config,
+  } = useCheckoutCommunication()
+  const reduxDispatch = useDispatch()
+  const [current, send] = useMachine(checkoutMachine)
 
-  useEffect(() => {
-    if (account && account.address) {
-      emitUserInfo({
-        address: account.address,
-      })
-    }
-  }, [account])
+  const paywallConfig = config || configFromSearch
 
   useEffect(() => {
-    if (!config && configFromSearch) {
-      dispatch(setConfig(configFromSearch))
+    if (account && paywallConfig) {
+      send('gotConfigAndAccount')
+    } else if (paywallConfig) {
+      setTimeout(() => send('gotConfig'), 500)
```

Comment:
This delay prevents a flicker of the not logged in view when we get the config just slightly before we get the account.

Reply:
Ha! We really need to change our approach to login users in.